.travis/complie_check.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+dist_list=$(go tool dist list)
+
+for dist in ${dist_list}; do
+    GOOS=$(echo ${dist} | cut  -d "/" -f 1)
+    GOARCH=$(echo ${dist} | cut -d "/" -f 2)
+    echo "Building  ${GOOS}/${GOARCH}"
+    GOOS=${GOOS} GOARCH=${GOARCH} go build  -o /dev/null
```

Comment:
if `go build fails`, does the test fail?
checking the exit code is needed.

add `set -e`

or

```bash
EXITCODE=0
# snip
GOOS=${GOOS} GOARCH=${GOARCH} go build  -o /dev/null || EXITCODE=1
# snip
exit "$EXITCODE"
```

Reply:
This should use `set -e` at the top of the script. We also want it to fail if e.g. `go tool dist list` fails or the cuts do not work as expected.